config/base/derive/src/proxy.rs

```diff
@@ -35,86 +37,115 @@ pub fn impl_proxy(ast: StructWithFields) -> TokenStream {
-                    #l_value = <#ty as From<#inner_ty>>::from(
-                        inner.map_err(|e| ::iroha_config_base::derive::Error::field_error(stringify!(#ident), e))?
-                    );
-                }
-            };
-
-            let inner_thing2 = if field.has_inner && field.has_option {
-                let inner_ty = get_inner_type("Option", ty);
-                let loadenv_trait = quote! { ::iroha_config_base::proxy::LoadFromEnv };
-                let err_variant = quote! { ::iroha_config_base::derive::Error::ProxyBuildError };
-                quote! {
-                    <#inner_ty as #loadenv_trait>::load_environment(#l_value.as_mut().ok_or(#err_variant(stringify!(#ident).to_owned()))?)?;
-                }
-            } else if field.has_inner {
-                quote! {
-                    #l_value.load_environment()?;
+                if field.has_inner {
+                    quote! {
+                        let #ident = if let Ok(var) = std::env::var(#field_env) {
+                            let inner: Result<#inner_ty, #err_ty> = #inner;
+                            inner.ok()
+                        } else { None };
+                        let inner_proxy = <#inner_ty as ::iroha_config_base::proxy::LoadFromEnv>::from_env();
+                        let #ident = if #ident.is_some() {
+                            Some(<#inner_ty as ::iroha_config_base::proxy::Combine>::combine(#ident.unwrap(), inner_proxy))
+                        } else {
+                            Some(inner_proxy)
+                        };
+                    }
+                } else {
+                    quote! {
+                        let #ident = if let Ok(var) = std::env::var(#field_env) {
+                            let inner: Result<#inner_ty, #err_ty> = #inner;
+                            inner.ok()
+                        } else { None };
+                    }
                 }
-            }
-            else {
-                quote! {}
             };
 
             quote! {
-                if let Ok(var) = std::env::var(#field_env) {
-                    #set_field
-                }
-                #inner_thing2
+                #set_field
             }
         });
 
     let name = &ast.ident;
+    let fields = ast
+        .fields
+        .iter()
+        .map(|field| {
+            let ident = &field.ident;
+            quote! { #ident }
+        })
+        .collect::<Vec<_>>();
+    let env_trait = quote! { ::iroha_config_base::proxy::LoadFromEnv };
     quote! {
-        impl ::iroha_config_base::proxy::LoadFromEnv for #name {
-            type Error = ::iroha_config_base::derive::Error;
-            fn load_environment(
-                &'_ mut self
-            ) -> core::result::Result<(), ::iroha_config_base::derive::Error> {
+        impl #env_trait for #name {
+            type ReturnValue = Self;
+            fn from_env() -> Self::ReturnValue {
```

Comment:
If we are always return `Self` does it make sense to simplify `LoadFromEnv` by removing `ReturnValue`?

Reply:
What if we're not? Moreover, this is just code fluff that gets optimised away after macro expansion...

Reply:
As @appetrosyan mentioned a long time ago, keeping an associated type allows to have more flexible impls in the future, if we ever need it to be done for something else.